From e6d44f410229a7c7dd0f05046cf1c86d9cbf7f41 Mon Sep 17 00:00:00 2001
Message-Id: <e6d44f410229a7c7dd0f05046cf1c86d9cbf7f41.1346940159.git.minovotn@redhat.com>
In-Reply-To: <c629acdc74c5b775c5cf7a7a3cb20ac833bcd0cc.1346940159.git.minovotn@redhat.com>
References: <c629acdc74c5b775c5cf7a7a3cb20ac833bcd0cc.1346940159.git.minovotn@redhat.com>
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 16 Aug 2012 11:39:12 +0200
Subject: [PATCH 10/18] Endian fix an assertion in usb-msd

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1345117160-21046-11-git-send-email-kraxel@redhat.com>
Patchwork-id: 40923
O-Subject: [RHEL-6.4 qemu-kvm PATCH 10/18] Endian fix an assertion in usb-msd
Bugzilla: 805172
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

This fixes a broken endian assumption in an assertion in usb-msd.

Cc: Gerd Hoffman <kraxel@redhat.com>

Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Signed-off-by: David Gibson <david@gibson.dropbear.id.au>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit e2854bf3239f57d160cfe5230033110c0c0d2837)

Conflicts:

	hw/usb-msd.c
---
 hw/usb-msd.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/usb-msd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/usb-msd.c b/hw/usb-msd.c
index d04b97f..8a8ced4 100644
--- a/hw/usb-msd.c
+++ b/hw/usb-msd.c
@@ -201,9 +201,9 @@ static void usb_msd_send_status(MSDState *s, USBPacket *p)
     int len;
 
     DPRINTF("Command status %d tag 0x%x, len %zd\n",
-            s->csw.status, s->csw.tag, p->len);
+            s->csw.status, le32_to_cpu(s->csw.tag), p->iov.size);
 
-    assert(s->csw.sig == 0x53425355);
+    assert(s->csw.sig == cpu_to_le32(0x53425355));
     len = MIN(sizeof(s->csw), p->len);
     memcpy(p->data, &s->csw, len);
     memset(&s->csw, 0, sizeof(s->csw));
@@ -240,7 +240,7 @@ static void usb_msd_command_complete(SCSIRequest *req, uint32_t status, int32_t
 
     s->csw.sig = cpu_to_le32(0x53425355);
     s->csw.tag = cpu_to_le32(req->tag);
-    s->csw.residue = s->residue;
+    s->csw.residue = cpu_to_le32(s->residue);
     s->csw.status = status != 0;
 
     if (s->packet) {
-- 
1.7.11.4

